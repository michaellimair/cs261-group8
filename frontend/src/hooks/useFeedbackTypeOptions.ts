import { FeedbackType } from 'customTypes/feedback';
import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';

const useFeedbackTypeOptions = () => {
  const { t } = useTranslation();

  return useMemo(() => [
    {
      label: t('dashboard.feedback.bug'),
      value: FeedbackType.BUG,
    },
    {
      label: t('dashboard.feedback.improvement'),
      value: FeedbackType.IMPROVEMENT,
    },
    {
      label: t('dashboard.feedback.question'),
      value: FeedbackType.QUESTION,
    },
  ], [t]);
};

export default useFeedbackTypeOptions;
