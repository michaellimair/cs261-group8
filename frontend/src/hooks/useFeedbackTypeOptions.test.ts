import { renderHook } from '@testing-library/react-hooks';
import useFeedbackTypeOptions from './useFeedbackTypeOptions';

describe('hooks/useFeedbackTypeOptions', () => {
  it('shows all the available options properly', () => {
    const { result } = renderHook(() => useFeedbackTypeOptions());

    expect(result.current).toMatchSnapshot();
  });
});
