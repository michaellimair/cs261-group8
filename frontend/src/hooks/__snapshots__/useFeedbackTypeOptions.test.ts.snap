// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks/useFeedbackTypeOptions shows all the available options properly 1`] = `
Array [
  Object {
    "label": "dashboard.feedback.bug",
    "value": "bug",
  },
  Object {
    "label": "dashboard.feedback.improvement",
    "value": "improvement",
  },
  Object {
    "label": "dashboard.feedback.question",
    "value": "question",
  },
]
`;
